### PR TITLE
Add support for Policies API and ITs

### LINKED ISSUE #250: Role.type should NOT be readOnly
This prevents role.setType() from being exposed

### PR DESCRIPTION
**Pull request by bdemers**
Also fixes #250

### CODE REVIEW

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/PolicyRulesIT.groovy [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+                    .setSession(client.instantiate(OktaSignOnPolicyRuleSignonSessionActions)
+                        .setUsePersistentCookie(false)
+                        .setMaxSessionIdleMinutes(720)
+                        .setMaxSessionLifetimeMinutes(0))))
+            .setConditions(client.instantiate(OktaSignOnPolicyRuleConditions)
+                .setNetwork(client.instantiate(PolicyNetworkCondition)
+                    .setConnection(PolicyNetworkCondition.ConnectionEnum.ANYWHERE))
+                .setAuthContext(client.instantiate(PolicyRuleAuthContextCondition)
+                    .setAuthType(PolicyRuleAuthContextCondition.AuthTypeEnum.RADIUS))
+                .setPeople(client.instantiate(PolicyPeopleCondition)
```
Comment (laura-rodriguez) [inline integration-tests/src/test/groovy/com/okta/sdk/tests/it/PolicyRulesIT.groovy]: AuthType should be `ANY` for this scenario.

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/PolicyRulesIT.groovy [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+
+class PolicyRulesIT implements CrudTestSupport {
+
+    private OktaSignOnPolicy crudTestPolicy
+
+    @Override
+    def create(Client client) {
+
+        crudTestPolicy = client.createPolicy(client.instantiate(OktaSignOnPolicy)
+            .setName("policy+" + UUID.randomUUID().toString())
```
Comment (laura-rodriguez) [inline integration-tests/src/test/groovy/com/okta/sdk/tests/it/PolicyRulesIT.groovy]: We agreed some time ago to add an SDK identifier to the resources to trace possible issues from the Okta Org.
Is it still valid @vijetmahabaleshwar-okta ?

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/PolicyRulesIT.groovy (hunk 1 above)
Comment (bdemers) [reply integration-tests/src/test/groovy/com/okta/sdk/tests/it/PolicyRulesIT.groovy]: 🤔 i would have expected this test to fail

Comment (bdemers) [reply integration-tests/src/test/groovy/com/okta/sdk/tests/it/PolicyRulesIT.groovy]: Doh, the test has a return type, so this test wasn't being run.  Looks like it was the only on though.
Good catch!

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/PolicyRulesIT.groovy (hunk 2 above)
Comment (bdemers) [reply integration-tests/src/test/groovy/com/okta/sdk/tests/it/PolicyRulesIT.groovy]: @laura-rodriguez @vijetmahabaleshwar-okta lets define this separately and I can find/replace these all (along with groups, apps, factors, etc)

**Pull request merged**